stripe2qbo/api/routers/transaction_router.py

```diff
@@ -0,0 +1,43 @@
+from typing import Annotated
+
+from sqlalchemy.orm import Session
+from fastapi import APIRouter, HTTPException, Depends
+
+from stripe2qbo.api.dependencies import get_current_user, get_db
+from stripe2qbo.db.models import TransactionSync, User
+
+router = APIRouter(
+    prefix="/transaction",
+    tags=["transaction"],
+)
+
+
+@router.get("/")
+async def get_all_transactions(
+    db: Annotated[Session, Depends(get_db)],
+    user: Annotated[User, Depends(get_current_user)],
+) -> TransactionSync:
+    try:
+        return (
+            db.query(TransactionSync).filter(TransactionSync.user_id == user.id).all()
+        )
+    except Exception as e:
```

Comment:
I'd like to avoid generic try/catch. Unless there is something specific you expect can break and throw an exception here,  you can remove the try/catch logic

DB connection or authentication errors should get caught by the dependencies. (These are Operational errors, that we expect might happen) Other than that, any other exception here would be a programming error. 

see https://levelup.gitconnected.com/distinction-between-operational-error-and-programmer-error-in-nodejs-bd77bca8da1 for the distinction